doc/image_proc.rst

```diff
@@ -0,0 +1,344 @@
+``UNITS``, which can be either ``'e-'`` or ``'ADU'``.
+
+Pattern-Noise Subtraction
+-------------------------
+
+.. Ryan, please check this.
+
+Some instruments, specifically Keck KCWI, are known to have a sinusoidal pattern
+in its bias.  :func:`~pypeit.images.rawimage.RawImage.subtract_pattern` models
+and subtracts this pattern from the data based on the overscan regions.  Unless
+you know such a pattern exists in your data, you should set the ``use_pattern``
+option to false (the default) for *all* frames.  Currently no error associated
+with this pattern subtraction is included in the image-processing error budget
+(unless the readnoise is determined empirically; see below).
+
+Read and Digitization Noise
+---------------------------
+
+Readnoise variance, :math:`V_{\rm rn}`, is calculated by
+:func:`~pypeit.core.procimg.rn2_frame`.  The calculation requires the detector
+readnoise (RN in elections, e-) and, possibly, gain (:math:`\gamma` in e-/ADU)
+for each amplifier, which are provided for each amplifier by the
+:class:`~pypeit.images.detector_container.DetectorContainer` object defined for
+each detector in each :class:`~pypeit.spectrographs.spectrograph.Spectrograph`
+subclass.  If a readnoise value is set to :math:`\leq 0`, the readnoise is
+estimated by the variance in the overscan regions of the image being processed;
+see :func:`~pypeit.images.rawimage.RawImage.estimate_readnoise`; use of the
+readnoise estimate regardless of the value provided by the
+:class:`~pypeit.images.detector_container.DetectorContainer` object can also be
+explicitly requested by setting the ``empirical_rn`` parameter to true.
+
+By default, the readnoise variance image does *not* include digitization noise.
+Digitization noise is driven by the conversion from counts to ADU via the gain
+and the quantization to an integer.  One can be derive the digitization noise by
+taking the second moment of a uniform distribution from -1/2 to 1/2 to find
+:math:`\sqrt{1/12}` ADU [1]_ [2]_, which is typically negligible (i.e., when the
+gain is on the same order as the readnoise).  And, in most cases, digitization
+noise will have been included in the estimate of the readnoise.  For this
+reason, digitization noise is *not* explicitly included in the ``PypeIt``
+variance model, and its inclusion cannot be turned on using a pypeit parameter.
+If you need to add digitization noise for your instrument, please `Submit an
+issue`_.
+
+Overscan Subtraction
+--------------------
+
+If available, the raw-image reader for each spectrograph returns the overscan
+region of the detector; see
+:func:`~pypeit.spectrographs.spectrograph.Spectrograph.get_rawimage`.  Overscan
+subtraction uses this region of the image to subtract a per-frame bias level;
+see :func:`~pypeit.core.procimg.subtract_overscan`.  Set the ``use_overscan``
+parameter to false if you do not want to use the overscan in this way or, more
+importantly, your instrument detector does not include an overscan region.
```

Comment:
If it doesn't, I should hope we have set this already in the spectrograph file.  Might edit this accordingly.

Reply:
Done.